mycroft/tts/mimic2_tts.py

```diff
@@ -124,31 +123,50 @@ def sentence_chunker(text, chunk_size, split_by_punc=True):
     Returns:
         list: list of text chunks
     """
-    text_list = text.split()
-    # if initial text is 1.3 times chunk size, no need to split
-    # if the chracter count is less then 55
-    if len(text_list) <= chunk_size * 1.3:
-        if len(text) < 55:
-            return [add_punctuation(text)]
+    if len(text) <= max_sentence_size:
+        return [add_punctuation(text)]
 
     # split text by punctuations if split_by_punc set to true
-    punc_splits = None
+    chunks = None
     if split_by_punc:
-        punc_splits = split_by_punctuation(text, chunk_size)
-
-    # split text by chunk size
-    chunks = []
-    if punc_splits:
-        for sentence in punc_splits:
-            sentence = sentence.strip()
-            chunks += split_by_chunk_size(sentence, chunk_size)
-    # split text by chunk size
-    else:
-        chunks += split_by_chunk_size(text, chunk_size)
+        # first split by periods ending puncs
+        chunks = split_by_punctuation(
+            text.strip(),
+            puncs=['.', '!', '?', ':', '-', ';']
+        )
+
+        # if sentence is still to big, split by other commas
+        second_splits = []
+        did_second_split = False
+        for sentence in chunks:
+            if len(sentence) > max_sentence_size:
+                comma_splits = split_by_punctuation(
+                    sentence.strip(), puncs=[',']
+                )
+                second_splits += comma_splits
+                did_second_split = True
+            else:
+                second_splits.append(sentence.strip())
+
+        if did_second_split:
+            chunks = second_splits
+
+        # if sentence is still to by 15 word chunks
```

Comment:
I think this is missing a couple of words and is i 15 word chunks or 20 word chunks?

```python
            chunk_split = split_by_chunk_size(sentence.strip(), 20)
```

Makes it look like 20